docs/docs/services/qix-engine/access-control.md

```diff
@@ -4,21 +4,40 @@
     This feature is in an experimental state. Use with caution
     since this feature may change in the future.
 
-Being able to control which users can do what in the context
-of a document is often necessary. In Qlik Associative Engine we support
-[Attribute-Based Access Control (ABAC)](https://en.wikipedia.org/wiki/Attribute-based_access_control)
-which lets you control the document resources based on a users'
-attributes rather than using roles or other, more static, means.
+Being able to control which users can do what in the context of a document is often necessary. In Qlik Associative
+Engine we support [Attribute-Based Access Control (ABAC)](https://en.wikipedia.org/wiki/Attribute-based_access_control)
+which lets you control the document resources based on a user's attributes rather than using roles or other more static
+means.
 
 ## Configuration
 
-Rules are defined, one on each row, in text files. Qlik Associative Engine will
-apply rules in order (except for deny rules which are always applied first,
-regardless of where they are defined).
+### Rules
 
-If the ABAC feature is enabled, all users will by default have no
-access, so it is important that you also include rules when enabling it
-or you will cause access problems for your users.
+Each rule is a conditional expression that evaluates to either `true` or `false`. Typically, a rule asserts if a _User_
+with certain attributes is allowed to perform a specific _Action_ on a certain _Resource_.
+
+ABAC rule evaluation is enabled in the engine by using the `-S EnableABAC=1` command line switch.
+
+!!! Note
+    If the ABAC feature is enabled, all users will by default have no access, so it is important that you also include
+    rules when enabling it or you will cause access problems for your users.
+
+### Rule Files
+
+Rules are defined in text files with one rule on each row. Qlik Associative Engine applies rules in the order they
+appear in the file. Two types of rule files are supported. The _Deny_ rule file and the _Allow_ rule file.
+
+The _Deny_ rule file denies access and is always evaluated first. If any rule in the _Deny_ file evaluates to `true`,
+access is immediately denied and no further processing of rules take place. The path to the _Deny_ rule file is passed
+to the engine using the `-S SystemDenyRulePath=<path to file>` command line switch.
+
+The _Allow_ rule file allows access. If any rule in the _Allow_ file evaluates to `true`, access is immediately granted
+and no further processing of rules take place. The path to the _Allow_ rule file is passed to the engine using the
+`-S SystemAllowRulePath=<path to file>` command line switch.
+
+### Example
+
+The following command shows how to start a Qlik Associative Engine instance as a Docker container, enabling ABAC
```

Comment:
In the examples repo, we set these switches:
`-S AcceptEULA=${ACCEPT_EULA} -S ValidateJsonWebTokens=2 -S JsonWebTokenSecret=secret -S DocumentDirectory=/docs -S SystemAllowRulePath=...`
I guess we don't need `PersistenceMode` and `Gen3` then? We don't even set `EnableABAC` which seems strange.

Reply:
I believe PersistenceMode=3 used to imply that ABAC is enabled too (if there was a file defined)

Reply:
Super-confusing as in the example we set neither of them and here both.
Maybe setting `-S SystemAllowRulePath=...` also implies `-S EnableABAC=1`?

Reply:
Yes, things have changed since we wrote this. You no longer need to use `PersistenceMode=3` to use ABAC, and showcasing it using `EnableABAC=1` seems correct to me.

Reply:
PersistenceMode 3 is not required for ABAC.